drivers/tests/TESTS/mbed_drivers/ticker/main.cpp

```diff
@@ -162,14 +156,14 @@ void test_case_2x_ticker()
 void test_multi_ticker(void)
 {
     Ticker ticker[TICKER_COUNT];
-    const uint32_t extra_wait = 5; // extra 5ms wait time
+    milliseconds extra_wait = 5ms; // extra 5ms wait time
 
     multi_counter = 0;
     for (int i = 0; i < TICKER_COUNT; i++) {
-        ticker[i].attach_us(callback(increment_multi_counter), MULTI_TICKER_TIME_MS * 1000);
+        ticker[i].attach(callback(increment_multi_counter), MULTI_TICKER_TIME);
     }
 
-    ThisThread::sleep_for(MULTI_TICKER_TIME_MS + extra_wait);
+    ThisThread::sleep_for(MULTI_TICKER_TIME.count() + extra_wait.count());
```

Comment:
This is unnecessary, and broken - you're adding a microsecond count to a millisecond count and then passing it to a deprecated non-Chrono API.

I guess the issue was that you were getting a compile error because you were passing microseconds to a millisecond API?

That could have been fixed by

     ThisThread::sleep_for(duration_cast<milliseconds>(MULTI_TICKER_TIME + extra_wait));

but it's actually easier to just not define MULTI_TICKER_TIME as microseconds in the first place. Make it `100ms`, then

    ThisThread::sleep_for(MULTI_TICKER_TIME + extra_wait);

Reply:
This is pretty neat! Does the chrono library handle the conversion when the function attempts to access it?

Reply:
I defined MULTI_TICKER_TIME in microseconds because of TickerBase::attach, however, defining it in terms of ms doesn't cause build errors like sleep_for. Is this because attach is overloaded by some distant base class which accepts milliseconds?

Reply:
The Chrono system does automatic conversions between resolutions where it's safe.

So if a function takes `microseconds`, and you pass it `milliseconds`, then it does the conversion, via a runtime or compile-time `*1000` on the count.

If the function takes `milliseconds` and you pass it `microseconds`, then you get a compile error due to the loss of precision.  To forcibly lose precision you have to explicitly do `duration_cast<milliseconds>(my_usec_thing)`, and that will then put in a `/1000`.

And this extends to conversions during the arithmetic operations: `5s + 1ms` = `5001ms`.

Basically if it compiles, and it's all Chrono, it does the right thing. The place you can make mistakes is when you use `count()` or constructors to move between Chrono and non-Chrono code, and you get the precision wrong.

Tutorial needed? https://www.youtube.com/watch?v=P32hvk8b13M

Reply:
Note that there's no leeway for "round constants" on the permitted conversions. It works purely on type, not value. You can't implicitly convert any microsecond value to milliseconds, not even a constant `1000us`.

If you do want a function that can accept arbitrary precision, then you can define it as a template that accepts any sort of `duration`, and you can choose how to do any rounding in the implementation. Functions in the standard C++ library work like this, eg https://en.cppreference.com/w/cpp/thread/sleep_for

But I chose not to do that in our Chrono conversion for simplicity.